crates/ursa-index-provider/src/provider.rs

```diff
@@ -190,16 +198,25 @@ where
                 current_head.map(|h| forest_ipld::Ipld::Link(convert_cid(h.to_bytes())));
             let sig = ad.sign(&keypair)?;
             ad.Signature = Ipld::Bytes(sig.into_protobuf_encoding());
-            let ipld_ad = forest_ipld::to_ipld(ad)?;
+            let ipld_ad = forest_ipld::to_ipld(&ad)?;
             let cid = bs.put_obj(&ipld_ad, Code::Blake2b256)?;
             *head = Some(cid);
             return Ok(());
         }
         return Err(anyhow!("ad not found"));
     }
 
-    async fn announce_msg(&self, peer_id: PeerId) -> Result<Vec<u8>> {
-        let msg_addrs = [self.config.domain.clone().parse().unwrap()].to_vec();
+    async fn create_announce_msg(&self, peer_id: PeerId) -> Result<Vec<u8>> {
+        let domain = match Multiaddr::from_str(&self.config.domain) {
```

Comment:
```suggestion
        let address = match Multiaddr::from_str(&self.config.domain)?;
```
just suggestion on the name, as we are returning a multiaddress not a domain